staker/block_validator.go

```diff
@@ -273,6 +278,9 @@ func (v *BlockValidator) readLastBlockValidatedDbInfo(reorgingToBlock *types.Blo
 		if genesisBlock == nil {
 			return fmt.Errorf("blockchain missing genesis block number %v", v.genesisBlockNum)
 		}
+		v.blockMutex.Lock()
+		defer v.blockMutex.Unlock()
```

Comment:
This defer won't hit until the end of the function, but we try to reacquire the blockMutex before that

Reply:
No that's fine, I just missed the return, sorry!